### DISCUSSION TITLE
XAML to C# Markup

### DISCUSSION THREAD

**Discussion by rvwilliams88** [unoplatform/uno.extensions#2331] [General] [answered]
I'm really struggling to convert XAML to C# Markup.  Please could someone help me by converting the following to C# Markup (from the PeopleApp sample)

````
<mvux:FeedView Source="{Binding People}">
    <DataTemplate>
        <ListView ItemsSource="{Binding Data}">

            <ListView.Header>
                <Button Content="Refresh" Command="{Binding Refresh}" />
            </ListView.Header>

            <ListView.ItemTemplate>
                <DataTemplate>
                    <StackPanel Orientation="Horizontal" Spacing="5">
                        <TextBlock Text="{Binding FirstName}"/>
                        <TextBlock Text="{Binding LastName}"/>
                    </StackPanel>
                </DataTemplate>
            </ListView.ItemTemplate>

        </ListView>
    </DataTemplate>
</mvux:FeedView>
```
`

**Comment by agneszitte** [comment, marked as answer]
Hi @rvwilliams88 Here is an example of a FeedView with ListView using C# Markup for the [TubePlayer sample app](https://github.com/unoplatform/Uno.Samples/tree/master/reference/TubePlayer):

https://github.com/unoplatform/Uno.Samples/blob/318e4215d9694fc4e10e952ac24d539c2d33ad43/reference/TubePlayer/src/TubePlayer/Presentation/MainPage.cs#L80
Here is the full related tutorial for this sample: https://platform.uno/docs/articles/external/workshops/tube-player/README.html

Let us know if with that you have a better idea of the C# Markup syntax or if you need additional support.

**Note:** Some of the samples in our documentation were created using XAML markup before we added support for C# Markup. We are continually improving our documentation to include more samples and examples for both XAML and C# Markup for comparison. 
Thank you for the feedback; it helps us identify and address these gaps.

(cc @dansiegel, @eriklimakc)

**Comment by rvwilliams88** [comment]
Thank you that was really useful. I now have the c# version running.  My next challenges are to change the list source to a database and then to replace the list view with a combo box. Finally I want to use the selected item to retrieve  and display related info from the database.
Regards
Richard

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by agneszitte** [reply]
@rvwilliams88 let us know if you need additional info or support for the other changes

**Comment by rvwilliams88** [comment]
Hi Agnès

Yes, I've hit a problem introducing a database for the source of 'people'.  See discussion 17072.

Regards

Richard.
****************************************************
Richard Williams
email: ***@***.******@***.***>
mobile: 07981 734697
****************************************************
